lib/phi_attrs/phi_record.rb

```diff
@@ -191,6 +206,16 @@ def __phi_stack
         RequestStore.store[:phi_access] ||= {}
         RequestStore.store[:phi_access][name] ||= []
       end
+
+      def __reset_phi_stack
+        RequestStore.store[:phi_access] ||= {}
```

Comment:
Why not just `RequestStore.store[:phi_access] = {}` ?

Reply:
Name is the class name so if you `Class1.disallow_phi!` you don't want to clear it on `Class2` if you had done both `Class1.allow_phi! and `Class2.allow_phi!` .